internal/app/referenceclient/impl.go

```diff
@@ -386,53 +350,35 @@ func (i *invoker) clientStream(
 		headers = internal.ConvertToProtoHeader(resp.Header())
 		trailers = internal.ConvertToProtoHeader(resp.Trailer())
 	}
-	var actualStatusCode int32
-	var connectErrorRaw *structpb.Struct
-	var actualTrailers []*v1.Header
-	wireDetails := getWireDetails(ctx)
-	if wireDetails != nil {
-		actualStatusCode = wireDetails.StatusCode
-		actualTrailers = wireDetails.Trailers
-		connectErrorRaw = wireDetails.ConnectErrorRaw
-	}
 
 	return &v1.ClientResponseResult{
-		ResponseHeaders:    headers,
-		ResponseTrailers:   trailers,
-		Payloads:           payloads,
-		Error:              protoErr,
-		ActualStatusCode:   actualStatusCode,
-		ActualHttpTrailers: actualTrailers,
-		ConnectErrorRaw:    connectErrorRaw,
+		ResponseHeaders:  headers,
+		ResponseTrailers: trailers,
+		Payloads:         payloads,
+		Error:            protoErr,
+		WireDetails:      getWireDetails(ctx),
 	}, nil
 }
 
 func (i *invoker) bidiStream(
 	ctx context.Context,
 	req *v1.ClientCompatRequest,
-) (result *v1.ClientResponseResult, _ error) {
+) (result *v1.ClientResponseResult, err error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	result = &v1.ClientResponseResult{
-		ConnectErrorRaw: nil, // TODO
-	}
+	result = &v1.ClientResponseResult{}
 
 	ctx = withWireCapture(ctx)
 
 	stream := i.client.BidiStream(ctx)
 	defer func() {
 		if result != nil {
```

Comment:
I'd invert this so you can un-indent the rest of the function. So basically this at the beginning:
```go
if err != nil {
    return
}
```